core/src/main/java/org/jruby/Ruby.java

```diff
@@ -1097,11 +1103,10 @@ public RubyModule defineModuleUnder(String name, RubyModule parent) {
         if (moduleObj != null ) {
             if (moduleObj.isModule()) return (RubyModule)moduleObj;
 
-            if (parentIsObject) {
-                throw newTypeError(moduleObj.getMetaClass().getName() + " is not a module");
-            } else {
-                throw newTypeError(parent.getName() + "::" + moduleObj.getMetaClass().getName() + " is not a module");
-            }
+            RubyString typeName = parentIsObject ?
+                    types(this, moduleObj.getMetaClass()) : types(this, parent, moduleObj.getMetaClass());
```

Comment:
maybe a `throw`is lost here - unless its intentional

Reply:
throw is immediately below that line.  It was two throws and now is one line since both throws just were assembling a proper typeName.